views/TodoList/TodoList.js

```diff
@@ -6,25 +6,53 @@ import Item from "../../components/Item/Item";
 import List from "../../components/List/List";
 import Loading from "../../components/Loading/Loading";
 
-export default function TodoList({ data, loading, error, insets }) {
-  // Use insets inside view components
+export default function TodoList({
+  data,
+  loading,
+  error,
+  insets,
+  onItemPress,
+}) {
   if (loading)
     return (
-      <View>
+      <View
+        style={{
+          flex: 1,
+          alignItems: "center",
+          justifyContent: "center",
+          paddingTop: insets.top,
+        }}
+      >
         <Loading />
       </View>
     );
   if (error)
     return (
-      <View>
+      <View
+        style={{
+          flex: 1,
+          alignItems: "center",
+          justifyContent: "center",
+          paddingTop: insets.top,
+        }}
+      >
         <Error message={error.message} />
       </View>
     );
   const header = <Header title="Todo List" />;
 
-  const renderItem = ({ item }) => <Item item={item} />;
+  const renderItem = ({ item }) => (
```

Comment:
we have a very useful util in the mobile-app codebase called generateComponent that I'd recommend implementing instead of creating these new mini components. Makes it much simpler to test and cleaner in your code